crypto/x509/x509_vfy.c

```diff
@@ -448,8 +448,8 @@ static int check_chain(X509_STORE_CTX *ctx)
 
     /*-
      *  must_be_ca can have 1 of 3 values:
-     * -1: we accept both CA and non-CA certificates, to allow direct
-     *     use of self-signed certificates (which are marked as CA).
+     * -1: we accept both CA and non-CA certificates, to allow direct use
```

Comment:
Do changes in x509_vfy.c really belong to this PR? IMO this wrongly mixes unrelated or only remotely related stuff into a single huge PR.

Reply:
I did those changes within this PR because otherwise three new tests would fail:
```
25-test_req.t .. 13/45 
CN = CA
error 79 at 0 depth lookup: invalid CA certificate
error self-signed_v1_CA_no_KIDs.pem: verification failed
../../util/wrap.pl ../../apps/openssl verify -x509_strict -trusted self-signed_v1_CA_no_KIDs.pem -partial_chain self-signed_v1_CA_no_KIDs.pem => 2
not ok 19 - strict verify allow self-signed_v1_CA_no_KIDs.pem
25-test_req.t .. 23/45 ---------------------------------------------------------
# self-signed_v3_CA_no_SKID.pem: OK
../../util/wrap.pl ../../apps/openssl verify -x509_strict -trusted self-signed_v3_CA_no_SKID.pem -partial_chain self-signed_v3_CA_no_SKID.pem => 0
not ok 26 - strict verify allow self-signed_v3_CA_no_SKID.pem
# ------------------------------------------------------------------------------
CN = EE
error 79 at 0 depth lookup: invalid CA certificate
error self-signed_v3_EE_wrong_keyUsage.pem: verification failed
../../util/wrap.pl ../../apps/openssl verify -x509_strict -trusted self-signed_v3_EE_wrong_keyUsage.pem -partial_chain self-signed_v3_EE_wrong_keyUsage.pem => 2
not ok 31 - strict verify allow self-signed_v3_EE_wrong_keyUsage.pem
# ------------------------------------------------------------------------------
25-test_req.t .. Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/45 subtests 
```

So I fear we need to keep them here.

Reply:
Meanwhile I've taken a pragmatic way out here: Simply turn those failing tests into TODOs
and move the improvements to the strict checks in `check_chain()` to a new PR: #13832.
I'll re-enable them there when this PR here is merged.